### PR TITLE
upgraded Xgboost4j to latest version

### PR DESCRIPTION
**Pull request by behrica**
one small code change was needed to specify n-rows in LabeledPoint

### CODE REVIEW

File: project.clj [hunk 1]
```diff
@@ -8,8 +8,10 @@ of how to use this library."
             :url "http://www.eclipse.org/legal/epl-v10.html"}
   :dependencies [[org.clojure/clojure "1.10.2-alpha1"]
                  [techascent/tech.ml.dataset "5.00-beta-25"]
-                 [ml.dmlc/xgboost4j "0.90"]
+                 [ml.dmlc/xgboost4j_2.12 "1.3.1"]
+                 [ml.dmlc/xgboost4j-spark_2.12 "1.3.1"]
                  [org.bytedeco/openblas "0.3.10-1.5.4"]
+                 [org.apache.hadoop/hadoop-common "3.3.0" :exclusions [org.apache.commons/commons-math3]]
```
Comment (behrica) [inline project.clj:14]: Dependency to hadoop  is required.

File: src/tech/v3/libs/xgboost.clj [hunk 2]
```diff
@@ -143,11 +144,10 @@
      "Multi-column regression/classification is not supported.  Target ds has %d columns"
      (ds/column-count target-ds))
     (map (fn [features target]
-           (LabeledPoint. (float target) nil (dtype/->float-array features)))
+           (LabeledPoint. (float target) (first (dtype/shape features))  nil (dtype/->float-array features)))
```
Comment (behrica) [inline src/tech/v3/libs/xgboost.clj:147]: This is the API change from 0.9 to 1.3.1

File: src/tech/v3/libs/xgboost.clj [hunk 3]
```diff
@@ -182,9 +182,9 @@
    :round (ml-gs/linear 5 46 5 :int64)
    :alpha (ml-gs/linear 0.01 0.31 30)})
 
-(defn ->dmatrix [feature-ds target-ds sparse-column]
+(defn ->dmatrix [feature-ds target-ds sparse-column n-sparse-columns]
```
Comment (behrica) [inline src/tech/v3/libs/xgboost.clj:185]: allowing to parse the n-sparse-columns by user fixes similar issue in xboost then i fixed #37

**Comment by cnuernber** [comment]
Huge upgrade.  Really nice work.  Hadoop is unfortunate but really not a problem.

**Pull request merged**